lms/static/scripts/frontend_apps/components/YouTubePicker.tsx

```diff
@@ -0,0 +1,86 @@
+import { Button, ModalDialog } from '@hypothesis/frontend-shared';
+import { useRef, useState } from 'preact/hooks';
+
+import { InvalidArgumentError } from '../errors';
+import {
+  validateYouTubeVideUrl,
+  videoIdFromYouTubeUrl,
+} from '../utils/youtube';
+import URLFormWithPreview from './URLFormWithPreview';
+
+export type YouTubePickerProps = {
+  /** The initial value of the URL input field. */
+  defaultURL?: string;
+
+  onCancel: () => void;
+  /** Callback invoked with the entered URL when the user accepts the dialog */
+  onSelectURL: (url: string) => void;
+};
+
+export default function YouTubePicker({
+  onCancel,
+  defaultURL,
+  onSelectURL,
+}: YouTubePickerProps) {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const [videoId, setVideoId] = useState(
+    defaultURL && videoIdFromYouTubeUrl(defaultURL)
+  );
+  const [error, setError] = useState<string>();
+
+  const verifyUrl = (inputUrl: string) => {
+    try {
+      const videoId = validateYouTubeVideUrl(inputUrl);
+
+      setVideoId(videoId);
+      setError(undefined);
+    } catch (e) {
+      const errorMessage =
+        e instanceof InvalidArgumentError
+          ? e.message
+          : 'Please enter a YouTube URL, e.g. "https://www.youtube.com/watch?v=cKxqzvzlnKU"';
+
+      setVideoId(undefined);
+      setError(errorMessage);
+    }
+  };
+  const confirmSelection = () => {
+    if (videoId) {
```

Comment:
Just checking the videoId is set for now. Once we add proper metadata fetching, we might need to check more properties